### PR TITLE
[DOC] Rewrite new PCA section

### LINKED ISSUE #547: More documentation for --tedpca
### Summary

Between version **0.0.7** and **0.0.8**, the `--tedpca` parameter changed.
Is it planned to enhance the documentation explaining which _tedpca_ does what ?

### Additional Detail

The "decision tree" part in the code is proably the most difficult part 

Help in **0.0.7** :
```
Possible choices: mle, kundu, kundu-stabilize
Method with which to select components in TEDPCA
Default: “mle”
```

Help in **0.0.8** :
```
Possible choices: kundu, kundu-stabilize, mdl, aic, kic
Method with which to select components in TEDPCA. PCA decomposition with the mdl, kic and aic options is based on a Moving Average (stationary Gaussian) process and are ordered from most to least aggresive. Default=’mdl’.
Default: “mdl”
```

### Next Steps

Explain a bit more what does each decision tree ?
In particular, why "mdl" is choose to be the default one ?

### PR DESCRIPTION
**Pull request by eurunuela**
Closes #547 .

Changes proposed in this pull request:

- Remove confusing lines.
- Mention why we have three options.
- Add links to Wikipedia for each option as it's probably the easiest way for users to have an idea of what each option does.
- Add a note that states it is a data-driven approach and the results might not be perfect (that's also why we provide three options).

### CODE REVIEW

File: docs/approach.rst
Comment (jbteves) [inline docs/approach.rst:253]: I think we probably want to link to the top of the article, rather than this subsection.

**Review by jbteves — CHANGES_REQUESTED** [review]
Thanks for adding these changes @eurunuela ! I have one minor request, but it's otherwise good.

File: docs/approach.rst
Comment (eurunuela) [reply docs/approach.rst:253]: Sure, whatever you think will be more meaningful for our users.

Comment (jbteves) [reply docs/approach.rst:253]: Is there something in particular you think would be valuable from that subsection? If so we could either add it to the docs or add a second link to it to highlight it.

Comment (eurunuela) [reply docs/approach.rst:253]: I wanted to highlight the relationship with AIC. I could link to the main page and then write a short sentence that says `you can see how it relates to AIC here.` with the link to that particular section.

Comment (jbteves) [reply docs/approach.rst:253]: That would be great!

Comment (eurunuela) [reply docs/approach.rst]: The latest commit deals with this. Thank you for the feedback!

File: docs/approach.rst
Comment (jbteves) [inline docs/approach.rst:222]:
```suggestion
  middle in terms of aggressiveness. You can see how KIC is related to AIC `here`_.
```

**Review by jbteves — CHANGES_REQUESTED** [review]
This looks good aside from a word skip. Once that's fixed I think this is ready to go.

File: docs/approach.rst
Comment (eurunuela) [reply docs/approach.rst:222]: My bad. Working on too many things at the same time. Thank you once again Josh!

**Review by jbteves — APPROVED** [review]
Looks great, thanks @eurunuela !

File: docs/approach.rst
Comment (tsalo) [inline docs/approach.rst:215]:
```suggestion
The number of selected principal components depends on the selection criteria.
```

File: docs/approach.rst
Comment (tsalo) [inline docs/approach.rst:217]:
```suggestion
to select the PCA components based on three widely-used model selection criteria:
```

File: docs/approach.rst
Comment (tsalo) [inline docs/approach.rst:225]: I think this should be at the top of this list, since it's the default.

File: docs/approach.rst
Comment (tsalo) [inline docs/approach.rst:232]: Could you add something like, "In addition to the moving average process-based approaches described above, we also support a decision tree-based selection method. This method, `kundu`, ..."

File: docs/approach.rst
Comment (tsalo) [inline docs/approach.rst:208]:
```suggestion
proposed by `Li et al (2007)`_ and used primarily in the Group ICA of fMRI Toolbox (GIFT).
```

**Review by tsalo — CHANGES_REQUESTED** [review]
This looks great. Thank you for doing it. I have a few minor ideas on how to provide slightly more context.

**Comment by eurunuela** [comment]
> This looks great. Thank you for doing it. I have a few minor ideas on how to provide slightly more context.

Thank you @tsalo ! Those were very helpful.

File: docs/approach.rst
Comment (tsalo) [inline docs/approach.rst:234]:
```suggestion
This method involves applying a decision tree to identify and discard PCA components which,
```

**Review by jbteves — APPROVED** [review]

**Review by tsalo — DISMISSED** [review]

**Review by jbteves — APPROVED** [review]

**Review by tsalo — APPROVED** [review]
We should really consider disabling the "dismiss stale reviews" rule. Anyway, this looks ready to merge. Once CI passes please merge and thanks again for doing this!

**Comment by eurunuela** [comment]
Thank you guys. I'm sorry I had to asked for a review every time.

**Pull request merged**
